source/YugiAcademy/YuGiOh/cards/card_management_system.py

```diff
@@ -0,0 +1,131 @@
+from django.core.exceptions import ObjectDoesNotExist
+
+from YuGiOh.cards import SpellCard, TrapCard, MonsterCard
+from assertions import SystemRestrictionInfringed, DataInconsistencyFound
+
+
+def raise_expected_to_be_found(managed_object):
+    raise DataInconsistencyFound(f'{managed_object} was expected to be found, but it was not.')
+
+
+def raise_found_card_named(name, card_type):
+    raise SystemRestrictionInfringed(f'There is already a {card_type} named {name}.')
+
+
+def raise_not_found_card_named(name, card_type):
+    raise SystemRestrictionInfringed(f'There is no {card_type} named {name}.')
+
+
+def card_named(name, card_type, repository, if_found, if_none):
+    try:
+        card = repository.get(name=name)
+        return card if if_found is None else if_found(card)
+    except ObjectDoesNotExist:
+        raise_not_found_card_named(name, card_type) if if_none is None else if_none()
```

Comment:
I don´t understand what you are doing. if if_none -> is None else if_none().
It´s not descriptive. It's a "trabalenguas"

Reply:
Already explained!